cms/djangoapps/maintenance/tests.py

```diff
@@ -0,0 +1,228 @@
+            login_url=reverse("login"),
+            original_url=url,
+        )
+
+        self.assertRedirects(response, redirect_url)
+
+
+class TestMaintenanceIndex(MaintenanceViewTestCase):
+    """
+    Tests for maintenance index view.
+    """
+
+    def setUp(self):
+        """Make the user global staff. """
+        super(TestMaintenanceIndex, self).setUp()
+        self.view_url = reverse('maintenance:maintenance')
+
+    def test_maintenance_index(self):
+        response = self.client.get(self.view_url)
+        self.assertContains(response, "Maintenance", status_code=200)
+
+        # Check that all the expected links appear on the index page.
+        for url in get_maintenace_urls():
+            self.assertContains(response, url, status_code=200)
+
+
+@ddt.ddt
+class TestForcePublish(MaintenanceViewTestCase, CourseKeyMixin):
+    """
+    Tests for the force publish view.
+    """
+
+    def setUp(self):
+        super(TestForcePublish, self).setUp()
+        self.view_url = reverse('maintenance:force_publish_course')
+        # Add some changes to course
+        chapter = ItemFactory.create(category='chapter', parent_location=self.course.location)
+        self.store.create_child(
+            self.user.id,
+            chapter.location,
+            'html',
+            block_id='html_component'
+        )
+        # verify that course has changes.
+        self.assertTrue(self.store.has_changes(self.store.get_item(self.course.location)))
+
+    def test_validate_error_messages(self):
+        """
+        Test all error messages for ForcePulish view.
+        """
+        # validate all course key error messags
+        self.verify_bad_course_keys()
+
+        # validate non split error message
+        course = CourseFactory.create(default_store=ModuleStoreEnum.Type.mongo)
+        self.verify_error_message(
+            {
+                'course-id': unicode(course.id)
+            },
+            _('Force publish course does not support old mongo style courses.')
+        )
+
+        # Validate course already published
+        response = self.force_publish_course()
+        self.validate_success_from_response(response, _('Now published and draft branch have same version'))
+
+        # now course is forcefully published, we should get already published course.
+        self.verify_error_message(
+            {
+                'course-id': unicode(self.course.id),
+                'dry-run': False
+            },
+            _('Course is already in published state.')
+        )
+
+    def force_publish_course(self, is_dry_run=''):
+        """
+        Force publish the course.
+        """
+        # get draft and publish branch versions
+        versions = get_course_versions(unicode(self.course.id))
+
+        # verify that draft and publish point to different versions
+        self.assertNotEqual(versions['draft-branch'], versions['published-branch'])
```

Comment:
Name suggests to me that method is going to force publish, but we are also checking versioning logic here. Can we extract it  and let the method do what it sounds like? Just a suggestion, I found it unrelated to check version in `force_publish_course` method.